tests/CMakeLists.txt

```diff
@@ -1,214 +1,18 @@
-)
-
-add_testpackage(TEST_NAME 
-                    motor_builder 
-                SOURCES 
-                    propulsion/motorbuilder_test.cpp
-                    ${CMAKE_SOURCE_DIR}/src/propulsion/motor_builder.cpp
-                TEST_INCLUDE_DIRECTORIES 
-                    ${CMAKE_SOURCE_DIR}/src/propulsion
-                    ${CMAKE_SOURCE_DIR}/src/types
-                    ${CMAKE_SOURCE_DIR}/tests/mock_libraries/propulsion
-)
-
-add_testpackage(TEST_NAME 
-                    spi_interface
-                SOURCES 
-                    spi/spi_interface_tests.cpp
-                TEST_INCLUDE_DIRECTORIES 
-                    ${CMAKE_SOURCE_DIR}/src/spi
-                    ${CMAKE_SOURCE_DIR}/src/types
-)
-
-add_testpackage(TEST_NAME 
-                    spi
-                SOURCES 
-                    spi/spi_tests.cpp
-                    ${CMAKE_SOURCE_DIR}/src/spi/spi.cpp
-                    ${CMAKE_SOURCE_DIR}/tests/mock_libraries/spi/gpio_config.c
-                    ${CMAKE_SOURCE_DIR}/tests/mock_libraries/spi/spi_config.c
-                    ${CMAKE_SOURCE_DIR}/tests/mock_libraries/spi/stm32g4xx_hal.c
-                TEST_INCLUDE_DIRECTORIES 
-                    ${CMAKE_SOURCE_DIR}/src/spi
-                    ${CMAKE_SOURCE_DIR}/src/types
-                    ${CMAKE_SOURCE_DIR}/tests/mock_libraries/spi
-)
-
-add_testpackage(TEST_NAME 
-                    imu_sensors_gyroscope
-                SOURCES 
-                    ${CMAKE_SOURCE_DIR}/src/imu/imu_sensor.cpp
-                    ${CMAKE_SOURCE_DIR}/src/imu/imu_sensor_with_sensitivity.cpp
-                    ${CMAKE_SOURCE_DIR}/src/imu/gyroscope.cpp
-                    ${CMAKE_SOURCE_DIR}/tests/imu/imu_gyroscope_tests.cpp
-                TEST_INCLUDE_DIRECTORIES 
-                    ${CMAKE_SOURCE_DIR}/src
-                    ${CMAKE_SOURCE_DIR}/src/imu
-                    ${CMAKE_SOURCE_DIR}/src/types
-                    ${CMAKE_SOURCE_DIR}/src/i2c
-                    ${CMAKE_SOURCE_DIR}/tests/mock_libraries/imu
-                    ${CMAKE_SOURCE_DIR}/tests/mock_libraries/i2c
-)
-
-add_testpackage(TEST_NAME 
-                    utilities_byte
-                SOURCES 
-                    ${CMAKE_SOURCE_DIR}/tests/utilities/utilities_byte_tests.cpp
-                TEST_INCLUDE_DIRECTORIES 
-                    ${CMAKE_SOURCE_DIR}/src
-)
-
-add_testpackage(TEST_NAME 
-                    imu_mpu9255
-                SOURCES 
-                    ${CMAKE_SOURCE_DIR}/src/imu/mpu9255.cpp
-                    ${CMAKE_SOURCE_DIR}/tests/imu/imu_mpu9255_tests.cpp
-                TEST_INCLUDE_DIRECTORIES 
-                    ${CMAKE_SOURCE_DIR}/src/imu
-                    ${CMAKE_SOURCE_DIR}/src/types
-                    ${CMAKE_SOURCE_DIR}/src/i2c
-                    ${CMAKE_SOURCE_DIR}/tests/mock_libraries/imu
-                    ${CMAKE_SOURCE_DIR}/tests/mock_libraries/i2c
-)
-
-SETUP_ALLTESTS_TARGET(
```

Comment:
Why exactly is this not necessary anymore? I just don't understand.

Reply:
This macro was just used, to combine all binaries. Those list of binaries was then used for the target "coverage" as dependency. If we have used "make all_tests" or "make coverage" in "build_unittests.sh" then all mentioned tests were executed.
This is now done by the ctest command in "run_all_unittests.sh". This is easier to maintain, because CMake (ctest) already knows all targets. So we do not have to bundle them before.
If you take a look in the build output, also the tests are still run 3 times and shuffled (and colored).

Reply:
All right! Got it. Nicely done!